In this project we will clean and give a preliminary analysis of a data set of used cars from the classified section of the German eBay website. This data set was originally scraped and uploaded to [Kaggle](https://www.kaggle.com/).

A sample of 50,000 points was taken. We will work tto clean up this data set.

In [1]:
import pandas as pd
autos=pd.read_csv('autos.csv',encoding='Latin-1')

In [2]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [3]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Most of the columns consist of the object data type indicating text data. Several columns have null values.

In [4]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [5]:
new_columns=['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_PS', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen']
autos.columns=new_columns

In [6]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_PS   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

We changed the columns headers from camel case to snake case. Now we'll do some elementary data exploration.

In [7]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-04-04 16:40:33  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_PS  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

Note that the price and odometer columns are stored as text. We can fix this by removing the '$', ',', and 'km' suffixes and renaming the columns to indicate unit of measurement. We will also convert  to a numeric type.

In [8]:
autos['registration_year'].value_counts().head()

2000    3354
2005    3015
1999    3000
2004    2737
2003    2727
Name: registration_year, dtype: int64

In [9]:
autos['price']=autos['price'].str.replace('$','').str.replace(',','').astype(int)


In [10]:
autos['price'].unique()

array([ 5000,  8500,  8990, ...,   385, 22200, 16995])

In [11]:
autos['odometer']=autos['odometer'].str.replace('km','').str.replace(',','').astype(int)

In [12]:
autos['odometer'].unique()

array([150000,  70000,  50000,  80000,  10000,  30000, 125000,  90000,
        20000,  60000,   5000, 100000,  40000])

In [13]:
autos.rename({'price':'price_dollars'},axis=1,inplace=True)
autos.rename({'odometer':'odometer_km'},axis=1,inplace=True)

In [14]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price_dollars',
       'abtest', 'vehicle_type', 'registration_year', 'gearbox', 'power_PS',
       'model', 'odometer_km', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

Now we'll analyze these columns more closely to look for ouliers.

In [15]:
autos['odometer_km'].unique().shape


(13,)

In [16]:
autos['price_dollars'].unique().shape

(2357,)

In [17]:
autos['odometer_km'].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [18]:
autos['price_dollars'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price_dollars, dtype: float64

In [19]:
autos['odometer_km'].value_counts()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64

We should note that odometer readings appear to be binned and in particualr any car with an odometer reading above 150,000 is rounded to just that. Indeed, more than half or the cars in the sample have odometer readings of 150,000 or greater.

In [20]:
autos['price_dollars'].value_counts().sort_index(ascending=False).tail()

5       2
3       1
2       3
1     156
0    1421
Name: price_dollars, dtype: int64

In [21]:
autos['price_dollars'].value_counts().sort_index(ascending=False).head(10)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
Name: price_dollars, dtype: int64

It seems unlikely that cars posted to ebay for near one million dollars or more would be very serious or representative entries.

In [22]:
autos['price_dollars'].mean(), autos['price_dollars'].std()

(9840.04376, 481104.38050019875)

We will remove any vehicles priced more  or less than 1.5 times the inter-quartile range from the lower and upper quartile's respectively. That is if $Q_1$is the 25% and $Q_3$ is the 75%, then outliers will be any data point outside the range $$[Q_1-1.5(Q_3-Q_1),\ Q_3+1.5(Q_3-Q_1)]$$

In [23]:
Q1=autos['price_dollars'].quantile(.25)
Q3=autos['price_dollars'].quantile(.75)
low_end=Q1-1.5*(Q3-Q1)
high_end=Q3+1.5*(Q3-Q1)

In [24]:
autos_clean=autos[autos['price_dollars'].between(low_end,high_end)]

In [25]:
autos_clean['price_dollars'].describe()

count    46216.000000
mean      3963.696101
std       3847.238683
min          0.000000
25%       1000.000000
50%       2500.000000
75%       5900.000000
max      16350.000000
Name: price_dollars, dtype: float64

Looks good. Now lets look at the date information.

In [26]:
autos_clean[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

The data in these 3 columns are formatted as strings. We wish to convert to numerical type to explore the data quantitatively.

Not that for the `date_crawled` column the first 10 characters give the day the day the car was sold. We'll remove everything but the date and then sort by index to determine the range of dates under consideration.

In [27]:
autos_clean['date_crawled'].value_counts(normalize=True,dropna=False).sort_index()

2016-03-05 14:06:30    0.000022
2016-03-05 14:06:40    0.000022
2016-03-05 14:07:04    0.000022
2016-03-05 14:07:08    0.000022
2016-03-05 14:07:21    0.000022
2016-03-05 14:07:26    0.000022
2016-03-05 14:07:40    0.000022
2016-03-05 14:07:45    0.000022
2016-03-05 14:08:00    0.000043
2016-03-05 14:08:05    0.000043
2016-03-05 14:08:27    0.000022
2016-03-05 14:08:42    0.000022
2016-03-05 14:09:02    0.000043
2016-03-05 14:09:20    0.000022
2016-03-05 14:09:22    0.000022
2016-03-05 14:09:38    0.000022
2016-03-05 14:09:46    0.000022
2016-03-05 14:09:56    0.000022
2016-03-05 14:09:57    0.000022
2016-03-05 14:09:58    0.000043
2016-03-05 14:10:18    0.000022
2016-03-05 14:10:20    0.000022
2016-03-05 14:11:03    0.000022
2016-03-05 14:11:05    0.000022
2016-03-05 14:11:14    0.000022
2016-03-05 14:11:15    0.000022
2016-03-05 14:11:25    0.000022
2016-03-05 14:11:40    0.000022
2016-03-05 14:11:42    0.000022
2016-03-05 14:11:56    0.000022
                         ...   
2016-04-

We see that the date range from March 5th to April 7th.

We can do the same with the `ad_created` and `last_seen` columns.

In [28]:
autos_clean['ad_created']=autos_clean['ad_created'].str[:10]
autos_clean['ad_created'].value_counts(normalize=True,dropna=False).sort_index()

/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


2015-08-10    0.000022
2015-09-09    0.000022
2015-11-10    0.000022
2015-12-05    0.000022
2015-12-30    0.000022
2016-01-03    0.000022
2016-01-07    0.000022
2016-01-10    0.000043
2016-01-13    0.000022
2016-01-16    0.000022
2016-01-22    0.000022
2016-01-27    0.000065
2016-01-29    0.000022
2016-02-01    0.000022
2016-02-02    0.000043
2016-02-05    0.000043
2016-02-08    0.000022
2016-02-09    0.000043
2016-02-11    0.000022
2016-02-12    0.000065
2016-02-14    0.000022
2016-02-16    0.000022
2016-02-17    0.000022
2016-02-18    0.000043
2016-02-19    0.000065
2016-02-20    0.000043
2016-02-21    0.000065
2016-02-22    0.000022
2016-02-23    0.000087
2016-02-24    0.000022
                ...   
2016-03-09    0.033279
2016-03-10    0.032370
2016-03-11    0.032716
2016-03-12    0.037065
2016-03-13    0.016769
2016-03-14    0.035702
2016-03-15    0.033711
2016-03-16    0.030314
2016-03-17    0.031418
2016-03-18    0.013675
2016-03-19    0.033279
2016-03-20    0.037693
2016-03-21 

In [29]:
autos_clean['last_seen'].str[:10].value_counts(normalize=True,dropna=False).sort_index()

2016-03-05    0.001168
2016-03-06    0.004674
2016-03-07    0.005647
2016-03-08    0.008071
2016-03-09    0.010321
2016-03-10    0.011252
2016-03-11    0.013177
2016-03-12    0.024926
2016-03-13    0.009456
2016-03-14    0.013069
2016-03-15    0.016271
2016-03-16    0.016942
2016-03-17    0.028843
2016-03-18    0.007638
2016-03-19    0.016358
2016-03-20    0.021270
2016-03-21    0.021270
2016-03-22    0.022135
2016-03-23    0.019149
2016-03-24    0.020209
2016-03-25    0.019971
2016-03-26    0.017397
2016-03-27    0.016596
2016-03-28    0.021659
2016-03-29    0.022979
2016-03-30    0.025532
2016-03-31    0.024364
2016-04-01    0.023628
2016-04-02    0.025186
2016-04-03    0.025835
2016-04-04    0.025186
2016-04-05    0.120867
2016-04-06    0.213671
2016-04-07    0.125281
Name: last_seen, dtype: float64

In [33]:
autos_clean['registration_year'].describe()

count    46216.000000
mean      2004.343864
std         96.225496
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2007.000000
max       9999.000000
Name: registration_year, dtype: float64

Lookin at the descriptive statistics for the `registration_year` column some of the values appear to be nonesense particularly looking at min and max values of year 1000 or 9999. This seems to be throwing off the standard deviation which at 96 years seems a bit to much. we could trim the data set by looking at a similiar Tukey range as above.

In [35]:
Q1=autos_clean['registration_year'].quantile(.25)
Q3=autos_clean['registration_year'].quantile(.75)
low_end=Q1-1.5*(Q3-Q1)
high_end=Q3+1.5*(Q3-Q1)

In [36]:
autos_clean=autos_clean[autos['registration_year'].between(low_end,high_end)]

/dataquest/system/env/python3/lib/python3.4/site-packages/ipykernel/__main__.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  if __name__ == '__main__':


In [37]:
autos_clean['registration_year'].describe()

count    45357.000000
mean      2003.411138
std          6.277579
min       1987.000000
25%       1999.000000
50%       2003.000000
75%       2007.000000
max       2019.000000
Name: registration_year, dtype: float64

It looks like doing this dropped about a 1000 data points from our sample but the data is much more reasonable, in particular the standard deviation. 

In [38]:
autos_clean['registration_year'].value_counts(normalize=True)

2000    0.073638
1999    0.065899
2005    0.065414
2003    0.059462
2004    0.059395
2001    0.059131
2006    0.057764
2002    0.055273
1998    0.053795
2007    0.047490
1997    0.044558
2008    0.044117
2009    0.040126
1996    0.031594
2017    0.031307
1995    0.028772
2016    0.028088
2010    0.027206
2011    0.024583
2012    0.017924
1994    0.014353
2018    0.010539
1993    0.009613
2013    0.008797
1990    0.008554
1992    0.008356
1991    0.007650
2014    0.005357
1989    0.003924
1988    0.003109
2015    0.002580
1987    0.001565
2019    0.000066
Name: registration_year, dtype: float64

Next we'll look at the brand data and see which brands are being put on ebay and for how much.

In [39]:
autos_clean['brand'].unique()

array(['peugeot', 'bmw', 'volkswagen', 'smart', 'ford', 'chrysler',
       'seat', 'renault', 'mercedes_benz', 'audi', 'opel', 'mazda',
       'mini', 'toyota', 'dacia', 'nissan', 'jeep', 'saab', 'volvo',
       'mitsubishi', 'fiat', 'skoda', 'subaru', 'sonstige_autos', 'kia',
       'citroen', 'porsche', 'hyundai', 'honda', 'daewoo', 'suzuki',
       'chevrolet', 'land_rover', 'jaguar', 'alfa_romeo', 'lada', 'rover',
       'trabant', 'daihatsu', 'lancia'], dtype=object)

In [43]:
brand_counts=autos_clean['brand'].value_counts(normalize=True)
print(brand_counts)

volkswagen        0.217188
opel              0.116917
bmw               0.103226
mercedes_benz     0.085059
audi              0.078202
ford              0.072933
renault           0.052362
peugeot           0.031836
fiat              0.027978
seat              0.019975
skoda             0.016381
mazda             0.016249
nissan            0.015786
smart             0.015433
citroen           0.015014
toyota            0.013030
hyundai           0.010384
volvo             0.009106
mitsubishi        0.008665
honda             0.008510
mini              0.007606
sonstige_autos    0.007452
kia               0.007165
alfa_romeo        0.006835
suzuki            0.006328
chevrolet         0.005247
chrysler          0.003858
dacia             0.002844
daihatsu          0.002778
subaru            0.002271
jeep              0.001896
daewoo            0.001742
saab              0.001698
rover             0.001499
porsche           0.001455
jaguar            0.001213
land_rover        0.001213
l

Note that Volkswagon accounts for more than 20% of all cars in the data set. This is not too suprising as its a German manufacturer and the data is taken from the German ebay site. Let's analyze the brands that acount for around 2% or more of the total cars and see what the average price listed for such brands is.

In [49]:
print(brand_counts.loc[brand_counts>.02])
top_brands=brand_counts.loc[brand_counts>.02].index

volkswagen       0.217188
opel             0.116917
bmw              0.103226
mercedes_benz    0.085059
audi             0.078202
ford             0.072933
renault          0.052362
peugeot          0.031836
fiat             0.027978
Name: brand, dtype: float64


In [55]:
common_brand_meanvalues={}
for brand in top_brands:
    brand_mean=autos_clean.loc[autos_clean['brand']==brand,'price_dollars'].mean()
    common_brand_meanvalues[brand]=brand_mean
print(common_brand_meanvalues)

{'audi': 5492.551170002819, 'opel': 2583.0033943051103, 'bmw': 5432.602520290474, 'renault': 2164.8084210526317, 'peugeot': 2884.0436288088645, 'ford': 2836.7028415961304, 'fiat': 2546.6832151300237, 'volkswagen': 3993.884377220587, 'mercedes_benz': 5045.601866251944}


Here we see that Audi cars are listed for the most on average, with BMW and Mercedes Benz not far behind and all above \$5000 dollars on average. Renault is listed for the least, with Ford, Opel, Peugot, and Fiat also averageing below \$3000 dollar listings. Volkswagon sits in the middle at around a \$4000 dollar average.

Next, we'll create a dataframe storing the mean price and mean mileage for each of these top brands.

In [56]:
brand_mean_mileage={}
for brand in top_brands:
    brand_mean=autos_clean.loc[autos_clean['brand']==brand,'odometer_km'].mean()
    brand_mean_mileage[brand]=brand_mean

In [61]:
bmp=pd.Series(common_brand_meanvalues)
bmm=pd.Series(brand_mean_mileage)
brand_price_km=pd.DataFrame(bmp,columns=['mean_price'])
brand_price_km['mean_odometer']=bmm
brand_price_km

mean_price  mean_odometer
audi           5492.551170  139705.384832
bmw            5432.602520  138738.786843
fiat           2546.683215  118337.273444
ford           2836.702842  127320.133011
mercedes_benz  5045.601866  138650.855365
opel           2583.003394  131001.320008
peugeot        2884.043629  128085.180055
renault        2164.808421  129218.947368
volkswagen     3993.884377  133375.291849

In [63]:
brand_price_km.sort_values(by='mean_price',ascending=False)

mean_price  mean_odometer
audi           5492.551170  139705.384832
bmw            5432.602520  138738.786843
mercedes_benz  5045.601866  138650.855365
volkswagen     3993.884377  133375.291849
peugeot        2884.043629  128085.180055
ford           2836.702842  127320.133011
opel           2583.003394  131001.320008
fiat           2546.683215  118337.273444
renault        2164.808421  129218.947368